test/test_point_inclusion.cpp

```diff
@@ -66,7 +66,14 @@ TEST(SpherePointContainment, Basic)
     EXPECT_FALSE(sphere.containsPoint(Eigen::Vector3d(0.00, 0.00, 1.01)));
 
     // near two-axis maximum
-    const double sq2 = sqrt(2) / 2;
+    // HACK: sqrt(2) / 2 is a problem due to rounding errors since the distance
+    //       is computed as 1.0000000000000002 . In such case we subtract an
+    //       epsilon.
+    double sq2 = sqrt(2) / 2;
+    if (sq2 * sq2 + sq2 * sq2 > 1)
+      sq2 -= std::numeric_limits<double>::epsilon();
+    ASSERT_LE(sq2 * sq2 + sq2 * sq2, 1.0);
```

Comment:
[`EXPECT_DOUBLE_EQ`](https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#floating-point-comparison) comes to rescue!

Reply:
Nope: EXPECT_DOUBLE_EQ doesn't help with this:  `EXPECT_SURF( sphere.containsPoint(Eigen::Vector3d(sq2 , sq2 , 0.00)));`

Reply:
OK. I see. As you use this "hacky" code twice, could you please make a small function from it.
```cpp
// split length of 1 into the largest number elem, such that sqrt(elem^2 + elem^2) <= 1
double elem = largest_component_for_length(1.0)
```